index.js

```diff
@@ -0,0 +1,661 @@
+  //.
+  //. > A value which has a Comonad must provide an `extract` method on itself.
+  var Comonad = Z.Comonad =
+  FantasyLandTypeClass('sanctuary-type-classes/Comonad',
+                       [Functor, Extend],
+                       {extract: [Value]});
+
+  //. ### Built-in types
+  //.
+  //. This package provides implementations of each method for each built-in
+  //. type which can satisfy the requirements of the corresponding type class.
+  //.
+  //. Function can satisfy the requirements of Functor, for example, so this
+  //. package implements a `fantasy-land/map` method for Function, to which
+  //. the exported `map` function dispatches when appropriate.
+  //.
+  //. | Type class      |   Array   | Function  |  Object   |  String   |
+  //. |:--------------- |:---------:|:---------:|:---------:|:---------:|
+  //. | [Setoid][]      |     ✔︎     |     ✔︎     |     ✔︎     |     ✔︎     |
+  //. | [Semigroup][]   |     ✔︎     |           |     ✔︎     |     ✔︎     |
+  //. | [Monoid][]      |     ✔︎     |           |     ✔︎     |     ✔︎     |
+  //. | [Functor][]     |     ✔︎     |     ✔︎     |     ✔︎     |           |
+  //. | [Bifunctor][]   |           |           |           |           |
+  //. | [Profunctor][]  |           |           |           |           |
+  //. | [Apply][]       |     ✔︎     |     ✔︎     |           |           |
+  //. | [Applicative][] |     ✔︎     |           |           |           |
+  //. | [Chain][]       |     ✔︎     |           |           |           |
+  //. | [ChainRec][]    |           |           |           |           |
+  //. | [Monad][]       |           |           |           |           |
+  //. | [Foldable][]    |     ✔︎     |           |           |           |
+  //. | [Traversable][] |     ✔︎     |           |           |           |
+  //. | [Extend][]      |           |           |           |           |
+  //. | [Comonad][]     |           |           |           |           |
+
+  var methods = {
+
+    //  Array#toString :: Array a ~> String
+    'Array#toString': function() {
+      return '[' + this.map(Z.toString).join(', ') + ']';
+    },
+
+    //  Array#equals :: Array a ~> Array a -> Boolean
+    'Array#fantasy-land/equals': function(other) {
+      return other.length === this.length &&
+             this.every(function(x, idx) { return equals(other[idx], x); });
+    },
+
+    //  Array#concat :: Array a ~> Array a -> Array a
+    'Array#fantasy-land/concat': function(xs) { return this.concat(xs); },
+
+    //  Array#empty :: Array a ~> Array a
+    'Array#fantasy-land/empty': function() { return []; },
+
+    //  Array#map :: Array a ~> (a -> b) -> Array b
+    'Array#fantasy-land/map': function(f) {
+      return this.map(function(x) { return f(x); });
+    },
+
+    //  Array#ap :: Array a ~> Array (a -> b) -> Array b
+    'Array#fantasy-land/ap': function(fs) {
+      var xs = this;
+      return chain(function(f) { return map(f, xs); }, fs);
+    },
```

Comment:
I just updated this to remove the nested `for` loops of the earlier implementation. :)